addons/sourcemod/scripting/szf/infected.sp

```diff
@@ -279,6 +282,62 @@ public Action Infected_DebrisTimerMoving(Handle hTimer, int iDebris)
 	return Plugin_Stop;
 }
 
+public void Infected_DebrisTimerInteractBuildings(int iDebris)
+{
+	if (!IsValidEntity(iDebris))
+		return;
+	//kill off.
+
+	if (!GetEntProp(iDebris, Prop_Send, "m_bAwake"))
+	{
+		RequestFrame(Infected_DebrisTimerInteractBuildings, iDebris);
+		return;
+	}
+
+	float vecVelocity[3];
+	SDKCall_GetVelocity(iDebris, vecVelocity);
+	float flSpeed = GetVectorLength(vecVelocity);
+	if (flSpeed < 100.0)
+	{
+		RequestFrame(Infected_DebrisTimerInteractBuildings, iDebris);
+		return;
+	}
+	
+	ScaleVector(vecVelocity, 0.2);
```

Comment:
Would replace `0.2` with `GetGameFrameTime() * 1.2` for a more accurate distance debris likely to travel during a frame, `* 1.2` is just for extra safety.